util/src/test/java/tech/pegasys/artemis/util/mikuli/G1PointTest.java

```diff
@@ -214,4 +214,27 @@ void succeedsWhenAttemptToDeserialiseWithBFlagAndAFlagTrueThrowsIllegalArgumentE
         IllegalArgumentException.class,
         () -> G1Point.fromBytesCompressed(Bytes.fromHexString(xInput)));
   }
+
+  @Test
+  void succeedsWhenDifferentPointsHaveDifferentHashcodes() {
+    G1Point point1 = G1Point.random();
+    G1Point point2 = G1Point.random();
+    // Ensure that we have two different points, without assuming too much about .equals
+    while (point1.ecpPoint().equals(point2.ecpPoint())) {
+      point2 = G1Point.random();
+    }
+    assert (point1.hashCode() != point2.hashCode());
```

Comment:
Better to use an `Assertions.assertEquals` method call rather than the `assert` keyword as the JVM asserts are often disabled.

Reply:
Oops - yes, was in Python mode briefly :-)